superset/assets/src/dashboard/actions/sliceEntities.js

```diff
@@ -64,22 +67,21 @@ export function fetchAllSlices(userId) {
 
           return dispatch(setAllSlices(slices));
         })
-        .catch(error =>
-          Promise.all([
-            dispatch(
-              fetchAllSlicesFailed(
-                error.error ||
-                  error.statusText ||
-                  t('Could not fetch all saved charts'),
+        .catch(errorResponse =>
+          getClientErrorObject(errorResponse).then(({ error }) =>
+            Promise.all([
```

Comment:
`Promise.all` seems unnecessary.
Is anybody expecting the result from the dispatches?
Might be better to return the errorObject, or no need to return at all.

```js
.catch(errorResponse => 
  getClientErrorObject(errorResponse).then(({ error }) => {
    dispatch(...);
    dispatch(...);
  });
```

```js
.catch(errorResponse => {
  const errorObjectPromise = getClientErrorObject(errorResponse);
  errorObject.then(({ error }) => {
    dispatch(...);
    dispatch(...);
  });
  return errorObjectPromise;
}
```